Lightyear/Position.idr

```diff
@@ -0,0 +1,71 @@
+-- ------------------------------------------------------------ [ Position.idr ]
+-- Module    : Lightyear.Position
+-- Description : Positioning in source files.
+--
+-- This code is distributed under the BSD 2-clause license.
+-- See the file LICENSE in the root directory for its full text.
+-- --------------------------------------------------------------------- [ EOH ]
+module Lightyear.Position
+
+import Data.String
+
+%default total
+%access export
+
+||| Generic representation of a position.
+public export
+record Position where
+  constructor MkPos
+  fname   : Maybe String
+  lineNo  : Nat
+  colNo   : Nat
+
+defaultPos : Maybe String -> Position
+defaultPos fname = MkPos fname 1 1
+
+namespace Generic
+  ||| Increment the position one character.
+  partial
+  increment : (tabwidth  : Nat)
+           -> (curr_pos  : Position)
+           -> (character : Char)
+           -> Position
+  increment _      (MkPos f l _) '\n' = MkPos f (S l) 1
+  increment twidth (MkPos f l c) '\t' = MkPos f l     nextTab
+    where
+      partial
+      nextTab : Nat
+      nextTab = ((c + 7) `div` twidth) * twidth + 1
```

Comment:
Shouldn't this be ```((c-1 + twidth) `div` twidth) * twidth + 1```?
1. `c-1` converts to 0-based units
2. `+ twidth` shifts one tab width
3. `div, *` rounds down to the closest tab stop to the left
4. `+ 1` converts back to 1-based units

This is the same as your code if `twidth = 8` but not otherwise.